operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfiguration.java

```diff
@@ -113,8 +113,5 @@ default ConfigurationService getConfigurationService() {
 
   @SuppressWarnings("unchecked")
   @Override
-  default Class<P> getResourceClass() {
-    return (Class<P>) Utils.getFirstTypeArgumentFromSuperClassOrInterface(getClass(),
-        ControllerConfiguration.class);
-  }
+  Class<P> getResourceClass();
```

Comment:
We should keep a default implementation to avoid breaking the API.

Reply:
Well, yes, althoug not really the API, it's the implementation. But we should remove it in the future, to make it more explicit that this is not used

Reply:
It is ok to do ti for next? (I mean a separate PR, wdyt?)

Reply:
No, this method should stay as it's needed by some implementations, Quarkus in particular. And it is breaking the API because implementations that were relying on the provided default implementation won't compile anymore with this change.

Reply:
reverted and commented it for now, but I guess we can adjust also quarkus in the future if it will be removed.

For now fine to have it there IMO. 

This PR should be ready now, also the adjusted spring boot part.

Reply:
I don't think that this will change in Quarkus because the resource class is resolved at build time and is therefore fixed then, not resolved at runtime, which is why we need this method to stay as-is.